src/Microsoft.Framework.Primitives/StringValues.cs

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.Primitives
+{
+    /// <summary>
+    /// Represents zero/null, one, or many strings in an efficient way.
+    /// </summary>
+    public struct StringValues : IList<string>
+    {
+        private static readonly string[] EmptyArray = new string[0];
```

Comment:
You might want to use Array.Empty. I thought @stephentoub added it a couple of month ago.